sklearn/tests/test_isotonic.py

```diff
@@ -485,6 +485,16 @@ def test_isotonic_dtype():
             assert res.dtype == expected_dtype
 
 
+def test_isotonic_mismatched_dtype():
```

Comment:
This would actually be a great test for all regressors, I reckon. We should include it in the common tests (sklearn/utils/estimator_checks.py). Would you like to contribute that in a later PR?

Reply:
I can do that. :)

BTW the space thing is added by PyCharm so it didn't remove the parenthesis.  My working PC is not at my hand so I cannot change it now. Will fix later if not merged.